stock_unique_order_per_location/models/stock_move_line.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    def _action_done(self):
+        res = super()._action_done()
+        self._check_same_order_at_destination()
+        return res
+
+    def _check_same_order_at_destination(self, location=None):
+        move_lines = self.filtered(
+            lambda line: line.move_id.picking_type_id.same_next_picking
+        )
+        for line in move_lines:
+            destination_location = location or line.location_dest_id
+            if not any(
+                quant.available_quantity for quant in destination_location.quant_ids
+            ):
+                continue
```

Comment:
This is not correct. Either there are quants available and this is an error.
Or there is no quant available and then you need to perform the second check on the move lines

Reply:
fixed